log_outgoing_requests/log_requests.py

```diff
@@ -30,6 +44,10 @@ def install_outgoing_requests_logging():
 
     def new_request(self, *args, **kwargs):
         self.hooks["response"].append(hook_requests_logging)
-        return self._lor_initial_request(*args, **kwargs)
+        try:
+            return self._lor_initial_request(*args, **kwargs)
+        except ConnectionError as e:
+            hook_requests_exception(e, *args, **kwargs)
+            raise e
```

Comment:
A context manager is a more elegant pattern here. And shouldn't this target the requests base exception class instead of this very specific error?